packages/foundation-models/README.md

```diff
@@ -17,9 +17,153 @@ $ npm install @sap-ai-sdk/foundation-models
+  `Input prompt tokens consumed: ${tokenUsage.prompt_tokens}\n` +
+  `Output text completion tokens consumed: ${tokenUsage.completion_tokens}\n`
+);
+
+```
+
+You can see that one can send multiple messages in a single request.
+This is useful in providing a history of the conversation to the model.
+
+Some parameters like `max_tokens` and `temperature` can be also be passed to the request to control the completion behavior.
+Refer to `AzureOpenAiChatCompletionParameters` interface for knowing more parameters that can be passed to the chat completion request.
+
+#### Obtaining a client using deployment ID
+
+In case you want to obtain the model by using the ID of your deployment on your own you can pass it instead of a model name:
+
+```TS
+import { AzureOpenAiChatClient } from '@sap-ai-sdk/foundation-models';
+
+const response = await new AzureOpenAiChatClient({ deploymentId: 'd1234' }).run({
+  messages: [
+    {
+      'role':'user',
+      'content': 'What is the capital of France?'
+    }
+  ]
+});
+```
+
+You could also pass the [resource group](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/resource-groups?q=resource+group) name to the client along with the deployment ID:
+
+```TS
+const client = new AzureOpenAiChatClient({ deploymentId: 'd1234' , resourceGroup: 'rg1234' })
+```
+
+### Usage of Azure OpenAI Embedding Client
+
+Use the `AzureOpenAiEmbeddingClient` to send embedding requests to an OpenAI model deployed in SAP generative AI hub.
+You can pass the model name as a parameter to the client, the sdk will implicitly fetch the deployment ID for the model from the AI Core service and use it to send the request.
+
+The deployment information which includes deployment ID and properties like model name and model version is also cached by default for 5 mins so that performance is not impacted by fetching the deployment information for every request.
+
+```TS
+import { AzureOpenAiEmbeddingClient } from '@sap-ai-sdk/foundation-models';
+
+const client = new AzureOpenAiEmbeddingClient('text-embedding-ada-002');
+const response = await client.run({
+  input: 'AI is fascinating'
+});
+const embedding = response.getEmbedding();
+```
+
+#### Obtaining a client using deployment ID
+
+In case you want to obtain the model by using the ID of your deployment on your own you can pass it instead of a model name:
+
+```TS
+import { AzureOpenAiEmbeddingClient } from '@sap-ai-sdk/foundation-models';
+
+const response = await new AzureOpenAiEmbeddingClient({ deploymentId: 'd1234' }).run({
+  messages: [
+    {
+      'role':'user',
+      'content': 'Where is the deepest place on earth located?'
+    }
+  ]
+});
+```
+
+You could also pass the [resource group](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/resource-groups?q=resource+group) name to the client along with the deployment ID:
+
+```TS
+const client = new AzureOpenAiEmbeddingClient({ deploymentId: 'd1234' , resourceGroup: 'rg1234' })
+```
```

Comment:
I'd leave this out and just mention that the same API as for chat client applies here too in like one sentence

Reply:
I just wanted to be complete if the user reads only the embedding client, will make the changes though.